cmd/appliance/upgrade/prepare.go

```diff
@@ -214,7 +214,10 @@ func prepareRun(cmd *cobra.Command, args []string, opts *prepareUpgradeOptions)
 	}
 	if !opts.forcePrepare {
 		var skip []openapi.Appliance
-		appliances, skip = appliancepkg.CheckVersions(ctx, *initialStats, appliances, targetVersion)
+		appliances, skip, err = appliancepkg.CheckVersions(ctx, *initialStats, appliances, targetVersion)
+		if err != nil {
+			return err
```

Comment:
Is this a fatal error? If so, consider updating the error message; since the error message string from `version.NewVersion` would not be very helpful for anyone other then the developer.

if its not a fatal error, don't `return err`, handle it.

based on this; https://github.com/appgate/sdpctl/blob/4988c37dd77ba66ad5e7116f89505e2ced022a37/pkg/appliance/checks.go#L174-L175 

it does not seems like its treated as severe error.

Reply:
I'm in two minds about that. On one hand, failing to parse a version doesn't seem that big of a deal, but on the other hand, we do not encourage downgrading at all, so from that perspective, it's pretty important that we know which version is being uploaded and treat it as fatal if it's a downgrade.